contracts/src/test/kotlin/com/r3/developers/apples/contracts/BasketOfApplesContractPackBasketCommandTest.kt

```diff
@@ -0,0 +1,109 @@
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertVerifies(transaction)
+    }
+
+    @Test
+    fun outputContractStateSizeNotOne() {
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(outputBasketOfApplesState)
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "This transaction should only output one BasketOfApples state")
+    }
+
+    @Test
+    fun blankDescription() {
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    "",
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    outputBasketOfApplesStateWeight,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "The output BasketOfApples state should have clear description of Apple product")
+    }
+
+    @Test
+    fun basketWeightIsZero() {
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    outputBasketOfApplesStateDescription,
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    0,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "The output BasketOfApples state should have non zero weight")
+    }
+
+    @Test
+    fun basketWeightIsNegative() {
+        val transaction = buildTransaction {
+            addCommand(AppleCommands.PackBasket())
+            addOutputState(
+                BasketOfApples(
+                    outputBasketOfApplesStateDescription,
+                    outputBasketOfApplesStateFarm,
+                    outputBasketOfApplesStateOwner,
+                    -1,
+                    outputBasketOfApplesStateParticipants
+                )
+            )
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "The output BasketOfApples state should have non zero weight")
+    }
+
+    @Test
+    fun missingCommand() {
+        val transaction = buildTransaction {
+            addOutputState(outputBasketOfApplesState)
+            addSignatories(outputBasketOfApplesStateParticipants)
+        }
+        assertFailsWith(transaction, "List is empty.")
```

Comment:
This isn't a particularly useful response when a command is missing.

Reply:
to be resolved in https://r3-cev.atlassian.net/browse/CORE-15212 :)

Reply:
there was no handling in contracts tutorial for missing inputs. it will be improved 👍!